### PR TITLE
Adds http and https endpoints to augur-state

### PR DESCRIPTION
**Pull request by brianosaurus**
Creates a server that listens on http and https ports with a JSON-RPC interface.

Connects that to our API routing framework.

### CODE REVIEW

File: packages/augur-state/package.json [hunk 1]
```diff
@@ -36,8 +36,12 @@
     "@augurproject/core": "2.0.0-0",
     "@augurproject/ethersjs-provider": "1.0.0-0",
     "@types/async": "2.4.1",
+    "@types/cors": "^2.8.4",
+    "@types/express": "^4.16.1",
+    "@types/helmet": "^0.0.43",
     "@types/web3": "1.0.18",
     "async": "2.6.2",
+    "cors": "^2.8.5",
```
Comment (adrake33) [inline packages/augur-state/package.json]: Just a reminder: these should have the caret removed.

File: packages/augur-test/package.json [hunk 2]
```diff
@@ -31,17 +31,22 @@
...
+    "@augurproject/state": "1.0.0-0",
     "ganache-core": "^2.4.0"
   },
   "devDependencies": {
     "@types/jest": "24.0.4",
     "@types/node": "10.12.29",
+    "@types/supertest": "^2.0.7",
     "jest": "22.4.4",
     "pouchdb-adapter-memory": "7.0.0",
+    "supertest": "^4.0.2",
```
Comment (adrake33) [inline packages/augur-test/package.json]: Here too.

**Review by adrake33 — APPROVED** [review]
LGTM

File: packages/augur-state/src/IsJsonRpcRequest.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export function isJsonRpcRequest(message: any): boolean {
+  if (message.jsonrpc !== "2.0") return false;
```
Comment (robert-e-davidson3) [inline packages/augur-state/src/IsJsonRpcRequest.ts:3]: You may also want to test that `message` isn't `undefined` or 'null' since then `message.jsonrpc` will give an error.

File: packages/augur-state/src/index.ts [hunk 4]
```diff
@@ -1,27 +1,21 @@
-export {API} from "./api/API";
-export {ServerController} from "./server/ServerController";
-export {Controller} from "./Controller";
+import RunWorker from "./Sync.worker";
 
-import RunWorker from "./run.worker";
+// start the web worker and move console.log over to paint in the browser
+console.log = (msg: string) => {
```
Comment (robert-e-davidson3) [inline packages/augur-state/src/index.ts]: Does this mess with `console.log` calls made elsewhere?

Comment (brianosaurus) [reply packages/augur-state/src/index.ts]: I left that there thinking it would be removed when augur-ui gets hooked up. However, you make a good point and I should just remove it.

File: packages/augur-test/tests/state/HTTPEndpoint.test.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  augur = await makeTestAugur(ACCOUNTS);
+  db = await mock.makeDB(augur, ACCOUNTS);
+}, 120000);
+
+test("HTTPEndpoint :: Responds to ping Json RPC Request ", async () => {
+  const api = new API(augur, db);
+  const app = HTTPEndpoint.createApp(api);  // fire up the endpoint
+
+  const result = await request(app).post("/").send({
+    jsonrpc: " 2.0",
```
Comment (robert-e-davidson3) [inline packages/augur-test/tests/state/HTTPEndpoint.test.ts]: There's a space in the `" 2.0"`. Is that intentional?

**Review by robert-e-davidson3 — APPROVED** [review]
This is really good code.

File: packages/augur-state/src/IsJsonRpcRequest.ts (hunk 3 above)
Comment (brianosaurus) [reply packages/augur-state/src/IsJsonRpcRequest.ts:3]: Added (even though github isn't showing that right now).
